src/markdown-pages/opentelemetry-masterclass/hands-on/workshop-dotnet.mdx

```diff
@@ -0,0 +1,902 @@
+3. `.AddTelemetrySdk()` to set some [conventional telemetry SDK attributes](https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/#telemetry-sdk). These are:
+
+    - `telemetry.sdk.name`
+    - `telemetry.sdk.language`
+    - `telemetry.sdk.version`
+
+</Step>
+
+<Step>
+
+Configure OpenTelemetry tracing:
+
+<>
+
+```csharp fileName=Program.cs
+using dotnet.Controllers;
+using OpenTelemetry.Trace;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+
+var resourceBuilder = ResourceBuilder
+    .CreateDefault()
+    .AddService("fibonacci-dotnet")
+    .AddTelemetrySdk();
+
+builder.Services.AddOpenTelemetryTracing(tracerProviderBuilder =>
+{
+   tracerProviderBuilder
+       .SetResourceBuilder(resourceBuilder)
+       .AddSource(FibonacciController.ActivitySourceName)
+       .AddAspNetCoreInstrumentation()
+       .AddOtlpExporter();
+});
+
+var app = builder.Build();
+
+app.MapControllers();
+
+app.Run();
+```
+
+</>
+
+<br />
+
+Here, you use `.AddOpenTelemetryTracing()` to inject a new `TracerProviderBuilder` into your `WebApplicationBuilder`. You configure your `TracerProviderBuilder` with:
+
+- The resource builder you created in the last step
+- A tracer for your application
+- Automatic instrumentation for ASP.NET Core
+- An OTLP exporter
+
+You named the tracer that you created with `.AddSource()` using a constant, called `ActivitySourceName`. You’ll create this in a later step. Before you do, you need to configure your OTLP exporter.
+
+</Step>
+
+<Step>
+
+In _dotnet/docker-compose.yaml_, configure your OTLP exporter:
+
+<>
+
+```yaml fileName=dotnet/docker-compose.yaml
+version: '3'
+services:
+  fibonacci-dotnet:
+    build: ./
+    ports:
+      - "8080:80"
+    environment:
+      OTEL_EXPORTER_OTLP_ENDPOINT: https://otlp.nr-data.net:4317
+      OTEL_EXPORTER_OTLP_HEADERS: "api-key=${NEW_RELIC_API_KEY}"
+```
+
+</>
+
+<br />
+
+The first variable you set configures the exporter to send telemetry data to https://otlp.nr-data.net:4317. This is our US-based OTLP endpoint. If you’re in the EU, use https://otlp.eu01.nr-data.net instead.
+
+The second variable passes the `NEW_RELIC_API_KEY` from your local machine in the `api-key` header of your OTLP requests. You set this environment variable in a previous lesson.
+
+Next, instrument the logic for your `/fibonacci` endpoint.
+
+</Step>
+
+<Step>
+
+In _Controller/FibonacciController.cs_, create a tracer:
+
+<>
+
+```csharp fileName=Controllers/FibonacciController.cs
+using System.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+
+namespace dotnet.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class FibonacciController : ControllerBase
+{
+    public const string ActivitySourceName = "FibonacciService";
+    private ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+
+    [HttpGet]
+    public IActionResult Get(long n)
+    {
+        try
+        {
+            return Ok(new
```

Comment:
The `<Tutorial>` component highlights diffs automatically throughout the course of the document. The reason this was highlighted is because the `{` was moved onto its own line. Fixed!